src/Ng/CommandHelpers.cs

```diff
@@ -284,24 +285,69 @@ public static StorageFactory CreateStorageFactory(IDictionary<string, string> ar
             else if (storageType.Equals("Azure", StringComparison.InvariantCultureIgnoreCase))
             {
                 string storageAccountName;
-                if (!arguments.TryGetValue("-storageAccountName", out storageAccountName))
+                string storageKeyValue;
+
+                bool useKeyVault = true;
+
+                string storageContainer;
+                if (!arguments.TryGetValue("-storageContainer", out storageContainer))
+                {
+                    TraceRequiredArgument("-storageContainer");
+                    return null;
+                }
+
+                string storageAccountNameFrame;
+                if (!arguments.TryGetValue("-storageAccountName", out storageAccountNameFrame))
                 {
                     TraceRequiredArgument("-storageAccountName");
                     return null;
                 }
 
-                string storageKeyValue;
-                if (!arguments.TryGetValue("-storageKeyValue", out storageKeyValue))
+                string storageKeyValueFrame;
+                if (!arguments.TryGetValue("-storageKeyValue", out storageKeyValueFrame))
                 {
                     TraceRequiredArgument("-storageKeyValue");
                     return null;
                 }
 
-                string storageContainer;
-                if (!arguments.TryGetValue("-storageContainer", out storageContainer))
+                string vaultName;
```

Comment:
please move all the KeyVault related initializations and arg parsing to a separate method